src/ray/common/task/task_spec.h

```diff
@@ -527,12 +529,17 @@ class WorkerCacheKey {
   /// worker. \param required_resources The required resouce.
   /// worker. \param is_actor Whether the worker will be an actor. This is set when
   ///         task type isolation between workers is enabled.
-  /// worker. \param iis_gpu Whether the worker will be using GPUs. This is set when
+  /// worker. \param is_gpu Whether the worker will be using GPUs. This is set when
   ///         resource type isolation between workers is enabled.
+  /// worker. \param is_ancestor_detached_actor Whether the worker will be running
+  ///         tasks or actors whose ancestor is a detached actor. This is set
+  ///         to prevent worker reuse between tasks whose ancestor is the driver process
+  ///         and tasks whose ancestor is a detached actor.
   WorkerCacheKey(const std::string serialized_runtime_env,
                  const absl::flat_hash_map<std::string, double> &required_resources,
                  bool is_actor,
-                 bool is_gpu);
+                 bool is_gpu,
+                 bool is_ancestor_detached_actor);
```

Comment:
Since this is a bool, does that mean workers can still be reused across different detached actors?

Also, `is_descended_from_detached_actor` is a bit easier to understand.

Reply:
> does that mean workers can still be reused across different detached actors?

Yes, this PR doesn't address the worker leak for detached actors. In the follow-up PR that fixes worker leak for detached actors, we won't allow worker reuse across different detached actors anymore.